src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,11 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    /**
+     * Check if command type mutates data in address book.
+     */
+    public boolean isMutating() {
```

Comment:
You can define this method as `abstract` and I think you can remove the exception.